### ISSUE TITLE
Adicionar mais testes

### ISSUE THREAD

**Issue by rdenadai** [milharal-dev/nemli-nemlerei-bot#41]
Aumentar a quantidade de testes unitários do bot.
Pode ser necessário algum tipo de refactor para melhor separação de código.

**Comment by luksquaresma** [comment]
> Aumentar a quantidade de testes unitários do bot. Pode ser necessário algum tipo de refactor para melhor separação de código.

Concordo,está bstante fragmentado e confuso. Por mim abrimos uma call e decidimos isso junto, porque vai ficar complicado fazer essa mudançá toda async.

**Comment by ryukinix** [comment]
Seria legal se conseguissemos fazer testes e2e (sistema), mas pode ficar pra outra etapa.
![image](https://github.com/user-attachments/assets/d689e3a2-dc25-40fe-9ad8-2c7c5578b7cd)
por enquanto só temos testes unitários, pra fazer testes de integração teríamos que emular openai/discord q são impossíveis basicamente, e como não temos bd, só vejo sentido em ter unit + e2e.

**Comment by rdenadai** [comment]
Eu imagino a os testes que fazem sentido seriam unit tests mesmo.
A ideia q pensei seria fragmentar códigos q estão ainda "grudados" demais em funções menores.
Assim a gente consegue testar + partes ...
